pkg/kvstore/allocator/allocator.go

```diff
@@ -287,10 +287,31 @@ func (k *kvstoreBackend) UpdateKey(ctx context.Context, id idpool.ID, key alloca
 	case success:
 		log.WithField(logfields.Key, keyPath).Warning("Re-created missing master key")
 	}
+	return nil
+}
+
+// UpdateKey refreshes the record that this node is using this key -> id
+// mapping. When reliablyMissing is set it will also recreate missing master or
+// slave keys.
```

Comment:
Ideally we would avoid using master/slave words here. Is there a better option?

Reply:
Agree. I already added a section in the PR description about this, and that we should probably come up with a more consistent and better naming. I think I would prefer doing that in a separate PR to differentiate between phrasing and logic, but if you think we should do it here, I'm happy to! I can do that when we agree on the overall idea here, and when we have a better phrasing.

I'm not 100% convinced what the best alternative phrasing is either, so very open to suggestions.
- `identity` and `node reference`?
- `leader` and `follower`?
- `primary` and `secondary`?

Reply:
I think that in this case, `identity` and `node reference` are probably the _most_ correct, if a bit long.

Reply:
Marking as resolved, given that this part has been dropped in the latest version.

Reply:
Thanks.

Lets follow up on the naming convention here in a follow up PR! :)